syz-manager/rpc.go

```diff
@@ -521,25 +521,26 @@ func (serv *RPCServer) doneRequest(runner *Runner, resp rpctype.ExecutionResult)
 	if req.try < resp.Try {
 		runner.logProgram(resp.ProcID, req.serialized)
 	}
-	if !serv.cfg.Cover {
-		addFallbackSignal(req.req.Prog, info)
-	}
 	for i := 0; i < len(info.Calls); i++ {
 		call := &info.Calls[i]
 		call.Cover = runner.instModules.Canonicalize(call.Cover)
 		call.Signal = runner.instModules.Canonicalize(call.Signal)
 	}
 	info.Extra.Cover = runner.instModules.Canonicalize(info.Extra.Cover)
 	info.Extra.Signal = runner.instModules.Canonicalize(info.Extra.Signal)
-	var err error
-	if resp.Error != "" {
-		err = fmt.Errorf("%s", resp.Error)
-	}
-	req.req.Done(&queue.Result{
+
+	result := &queue.Result{
+		Status: queue.Success,
 		Info:   info,
 		Output: resp.Output,
-		Err:    err,
-	})
+	}
+	if resp.Error != "" {
+		result.Status = queue.ExecFailure
+		result.Err = fmt.Errorf("%s", resp.Error)
+	} else if req.req.ExecOpts.ExecFlags&ipc.FlagCollectSignal > 0 {
+		addFallbackSignal(req.req.Prog, info)
```

Comment:
Don't we need to add it only if `!serv.cfg.Cover`?

Btw, we now still set ipc.FlagCollectSignal in pkg/fuzzer even if !serv.cfg.Cover, right? Previously we didn't. Can it confuse some code?

Reply:
Good point!
We ignore cfg.Cover in `pkg/fuzzer` and `pkg/vminfo`.

Reply:
Regarding `pkg/fuzzer` -- actually, another good question is what `serv.cfg.Cover` **should** mean for the `fuzzer.Fuzzer`. It's coverage-guided anyway, isn't it? The question is only whether we get signal from real PCs or our signal is based on syscall number + errno. The only place where it does make a difference is to determine the distribution of `exec fuzz` vs `exec gen` requests.

The `FlagCollectSignal` flag itself is just a means for the fuzzer to say `for these requests (where it's set) I want to get some signal feedback, for others (if it's not set), I don't care`. So I think we should not change anything in this regard. The existing fuzzer code is fine.

In `rpc.go` (or, maybe, it's easier right in `pkg/ipc`) we should probably not pass `FlagCollectSignal` to the executor if `EnvFlags&ipc.FlagSignal == 0`.

---

In `pkg/vminfo`, it's more interesting. It makes 0 sense to e.g. check `FeatureDelayKcovMmap` or `FeatureExtraCoverage` if `serv.cfg.Cover=false`.

Reply:
Fixed the `!serv.cfg.Cover` bug.